### ISSUE TITLE
MultiSearchParameters missing vector_query field

### ISSUE THREAD

**Issue by tanczosm** [DAXGRID/typesense-dotnet#145]
With the explosion of GPT related apps I was looking to see if this library could support vector searches and for the most part it looked good with just a small omission.

For vector searches the MultiSearchParameters class is missing a vector_query field.  The vector query is a list of floats along with a *k* value denoting the number of documents to return.  There can also be an "id" portion of the vector query if you want to find documents that are similar to an existing document.  Furthermore, there can also be a "flat_search_cutoff" parameter for brute-force searching purposes.

See [Nearest Neighbor Vector Search](https://typesense.org/docs/0.24.1/api/vector-search.html#creating-a-collection-with-a-vector-field)

Here is a javascript sample query:

```javascript
let searchRequests = {
  'searches': [
    {
      'collection': 'docs',
      'q': '*',
      'vector_query' : 'vec:([0.96826, 0.94, 0.39557, 0.306488], k:100)'
    }
  ]
}
let commonSearchParams = {}
client.multiSearch.perform(searchRequests, commonSearchParams)
```

Here is a sample response:
```javascript
{
  "facet_counts": [],
  "found": 1,
  "hits": [
    {
      "document": {
        "id": "0",
        "vec": [
          0.04, 0.234, 0.113, 0.001
        ]
      },
      "highlight": {
        "full": {},
        "snippet": {}
      },
      "highlights": [],
      "vector_distance": 0.19744956493377686
    }
  ],
  "out_of": 1,
  "page": 1,
  "request_params": {
    "collection_name": "docs",
    "per_page": 10,
    "q": "*"
  },
  "search_cutoff": false,
  "search_time_ms": 0
}
```

**Comment by runeanielsen** [comment]
Hi @tanczosm,

Thanks for creating the issue. I'll take a look when I get the time.

// Rune

**Comment by runeanielsen** [comment]
Hi @tanczosm,

Is it fine if I just make the `vector_query` a string? I might implement a helper function in the future, that can convert a typed C# object into a string query.

**Comment by tanczosm** [comment]
That is totally fine with me.

**Comment by runeanielsen** [comment]
Hi @tanczosm,

I've created a pull request with support for vector search. The pull-request contains an example showcasing how the API works, do you mind taking a look and check if it supports all the requirements you have, before I merge it in?

**Comment by tanczosm** [comment]
I updated PR #146 with some additional code for VectorQuery objects.  After giving it some more thought, given how common it will be to work with embeddings as arrays of floats there definitely is going to be a need for something that can encode the search query.

I included code to make all that possible.

**Comment by runeanielsen** [comment]
@tanczosm I can see in your example that you've implemented it using a `float` do you know what level of precision is needed, the reason I am asking is if we would ever be in a situation where `double` precision is needed?

**Comment by tanczosm** [comment]
The Typesense implementation is an array of float values, which is probably
all that matters.  I haven't seen anything but floats and all the
embeddings I pulled from OpenAI are also floats.

**Comment by runeanielsen** [comment]
Ok, I'll switch to floats then :+1:

**Comment by tanczosm** [comment]
Resolved in #146

**Comment by runeanielsen** [comment]
Hi @tanczosm, thanks for the contributions, I've released a new version 6.4.0 that contains the changes. :)
